### PR TITLE
Adding openshift plugin with release

### PR DESCRIPTION
**Pull request by JaydipGabani**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: Tidy
+        env:
+          GOPROXY: "https://proxy.golang.org"
+        run: cd main && go mod tidy && git diff --quiet HEAD
+
+      - name: Build openshift plugin
+        env:
+          GOPROXY: "https://proxy.golang.org"
+        run: cd main/ && GOOS=linux GOARCH=amd64 go build -o ~/main/bin/OpenshiftPlugin-${{ github.event.inputs.version }} .
```
Comment (shawn-hurley) [inline .github/workflows/release.yml:32]: Do we need to build for more than just linux/amd64?

File: cmd.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+type openshiftOptionalFields struct {
+	StripDefaultPullSecrets bool
+	PullSecretReplacement   map[string]string
+	RegistryReplacement     map[string]string
+}
+
+func getOptionalFields(extras map[string]string) (openshiftOptionalFields, error) {
+	var fields openshiftOptionalFields
+	var err error
+	if len(extras["StripDefaultPullSecrets"]) > 0 {
```
Comment (shawn-hurley) [inline cmd.go:46]: Potentially move these to constants and use in both places?

File: .github/workflows/release.yml (hunk 1 above)
Comment (JaydipGabani) [reply .github/workflows/release.yml:32]: We need to figure out which ones we want to provide the binaries for.

File: .gitignore [hunk 3]
```diff
@@ -12,4 +12,4 @@
 *.out
 
 # Dependency directories (remove the comment below to include it)
-# vendor/
+vendor/
```
Comment (sseago) [inline .gitignore:15]: We shouldn't be using vendor, so we probably don't need to add it to gitignore.

Comment (JaydipGabani) [reply .gitignore]: :+1:

File: cmd.go (hunk 2 above)
Comment (JaydipGabani) [reply cmd.go]: :+1:

File: cmd.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+)
+
+var logger logrus.FieldLogger
+
+const (
+	// flags
+	StripDefaultPullSecrets = "strip-default-pull-secrets"
+	PullSecretReplacement   = "pull-secret-replacement"
+	RegistryReplacement     = "registry-replacement"
+
```
Comment (sseago) [inline cmd.go:19]: Exchange lines 19 and 20, as in the latest version of my PR.

File: cmd.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+)
+func main() {
+	logger = logrus.New()
+	// TODO: add plumbing for logger in the cli-library and instantiate here
+	fields := []transform.OptionalFields{
+		{
+			FlagName: StripDefaultPullSecrets,
+			Help:     "Whether to strip Pod and BuildConfig default pull secrets (beginning with builder/default/deployer-dockercfg-) that aren't replaced by the map param pull-secret-replacement",
```
Comment (sseago) [inline cmd.go:27]:
```
			Help:     "Whether to strip Pod and BuildConfig default pull secrets (beginning with builder/default/deployer-dockercfg-) that aren't replaced by the map param " + PullSecretReplacement,
```

**Review by sseago — APPROVED** [review]
Approved, but with a couple minor changes to make. Basically these are the same changes @djzager requested on my PR and were updated on my last iteration.

**Review by sseago — APPROVED** [review]

**Pull request merged**